zstacklib/zstacklib/utils/rangeset.py

```
@@ -0,0 +1,173 @@
+    """
+    Non-overlapping merged half-open intervals [l, r) with integer endpoints.
+    Stores as a sorted list of (l, r) by l. Supports add, remove, membership,
+    total covered length, iteration, and computing missing gaps within [0, total).
+    """
+    __slots__ = ('iv',)
+
+    def __init__(self, iterable=None):
+        # Sorted by start: [(l1, r1), (l2, r2), ...], with r_i <= l_{i+1}
+        self.iv = []
+        if iterable:
+            for l, r in iterable:
```

Ruff (0.14.1)

Ambiguous variable name: `l`

(E741)